src/lib/y2storage/disk_analyzer.rb

```diff
@@ -203,140 +147,143 @@ def candidate_disks
     # MBR-based the MBR gap is nil, meaning "gap not applicable" which is
     # different from "no gap" (i.e. a 0 bytes gap).
     #
-    # @see #scope
+    # @param *disks [Disk, String] disks to analyze. All disks by default.
+    # @return [<Array<(DiskSize, nil)>] the value is the DiskSize of the
+    # MBR gap (or nil)
+    def mbr_gaps(*disks)
+      data_for(*disks, :mbr_gap) { |d| find_mbr_gap(d) }
+    end
+
+    # Variant of #mbr_gaps for an specific disk.
+    # @see #mbr_gaps
     #
-    # @return [Hash{String => (Y2Storage::DiskSize, nil)}] each key is the name
-    # of a disk, the value is the DiskSize of the MBR gap (or nil)
-    def mbr_gap
-      @mbr_gap ||= begin
-        result = find_mbr_gap
-        log.info("Found MBR gaps: #{result}")
-        result
-      end
+    # @param disk [Disk]
+    # @return [DiskSize, nil]
+    def mbr_gap(disk)
+      mbr_gaps(disk).first
     end
 
     # Partitions containing an installation of MS Windows
     #
     # This involves mounting any Windows-like partition to check if there are
     # some typical directories (/windows/system32).
     #
-    # @see #scope
-    #
-    # @return [Hash{String => Array<Partition>}] see {#partitions_with_id}
-    def windows_partitions
-      @windows_partitions ||= begin
-        result = find_windows_partitions
-        log.info("Found Windows partitions: #{result}")
-        result
+    # @param *disks [Disk, String] disks to analyze. All disks by default.
+    # @param check_limit [Fixnum] maximum number of disks to check.
+    # @return [Array<Partition>}] see {#partitions_with_id}
+    def windows_partitions(*disks, check_limit: DEFAULT_CHECK_LIMIT)
+      data_for(*disks, :windows_partitions, limit: check_limit) do |d|
+        find_windows_partitions(d)
       end
     end
 
-  private
-
-    attr_reader :devicegraph
-
-    # Set of disks to be used in most search operations
+    # Disks that are suitable for installing Linux.
     #
-    # @return [Array<String>]
-    def scoped_disks
-      @scoped_disks ||= scope == :install_candidates ? candidate_disks : dev_names(all_disks)
+    # @param check_limit [Fixnum] maximum number of disks to check
+    # for installation disks.
+    # @return [Array<Disk>] candidate disks
+    def candidate_disks(check_limit: DEFAULT_CHECK_LIMIT)
+      return @candidate_disks if @candidate_disks
+      @candidate_disks = find_candidate_disks(check_limit)
+      log.info("Found candidate disks: #{@candidate_disks}")
+      @candidate_disks
     end
 
-    # Array with all disks in the devicegraph
+    # Checks if a partition belongs to any VG or to an specefic one.
     #
-    # @return [Array<Disk>]
-    def all_disks
-      devicegraph.disks
+    # @param partition [Partition]
+    # @param vg [LvmVg] volume group
+    def partition_in_vg?(partition, vg = nil)
```

Comment:
Maybe some of this logic about a certain device being part of LVM should be moved to the BlkDevice class or similar places.